apps/src/templates/instructions/TopInstructions.jsx

```diff
@@ -556,7 +556,8 @@ class TopInstructions extends Component {
       isRtl ? styles.mainRtl : styles.main,
       mainStyle,
       {
-        height: height - RESIZER_HEIGHT
+        height: height - RESIZER_HEIGHT,
+        zIndex: 1021
```

Comment:
I'm curious, is this an arbitrary z index? Do we have guidelines on what values to use or what the boundaries of z indexes are?

Reply:
I've updated this to show where it comes from (borrowed from `InstructionsCSF`) and only set it when the overlay is showing beneath a dynamic instruction, which prevents the broader regression that @bethanyaconnor caught.  I'm not familiar with central guidelines, but there are occasional comments such as [here](https://github.com/code-dot-org/code-dot-org/blob/812322ed5b81643ec2cc28af768846a91b5e4868/apps/src/templates/Overlay.jsx#L9-L11).  Thanks go to you and @bethanyaconnor for the great catches.